tutorials/zkSync-RedStone-stable-price-marketplace-tutorial/README.md

```diff
@@ -0,0 +1,402 @@
+  For being able to use redstone oracles data, more inf:
+  https://docs.redstone.finance/docs/smart-contract-devs/get-started/redstone-core#1-adjust-your-smart-contracts 
+*/
+contract StableMarketplace is Marketplace, MainDemoConsumerBase {
+  /*
+    `_getPriceFromOrder` function should uses the `getOracleNumericValueFromTxMsg` function,
+    which fetches signed data from tx calldata and verifies its signature
+  */ 
+  function _getPriceFromOrder(
+    SellOrder memory order
+  ) internal view override returns (uint256) {
+    uint256 ethPrice = getOracleNumericValueFromTxMsg(bytes32("ETH"));
+    return (order.price / ethPrice) * (10 ** 8);
+  }
+}
+```
+
+#### 2. Adjust dApp TypeScript code
+
+The second thing to do is adjust the Typescript code of the dApp. Please take a look at the `blockchain.ts` file. Here you can find all functions required to make the marketplace work. But the function `buy` is not implemented. Here we will call the function from the contracts which require price data. To make it possible we need to wrap the contract instance with the [RedStone framework](https://github.com/redstone-finance/redstone-oracles-monorepo/tree/main/packages/evm-connector). If you are not familiar with our core model please read how to [adjust Typescript code](#2-adjust-javascript-code-of-your-dapp). After wrapping the contract we will be able to call the `getPrice` function from the `StableMarketplace` contract which eventually will call overridden `_getPriceFromOrder`. Now we are able to call the `buy` function from the `StableMarketplace` contract with the expected ETH amount to buy the NFT. Full implementation can be seen below:
+
+```js
+import { WrapperBuilder } from "@redstone-finance/evm-connector";
+
+async function getContractInstance(contractName) {
+  ...
+  return new ethers.Contract(address, abi, signer);
+}
+
+async function buy(orderId) {
+  const marketplace = await getContractInstance("marketplace");
+
+  // Wrapping marketplace contract instance.
+  // It enables fetching data from redstone data pool
+  // for each contract function call
+  try {
+    const wrappedMarketplaceContract = WrapperBuilder.wrap(
+      marketplace
+    ).usingDataService(
+      {
+        dataServiceId: "redstone-main-demo",
+        uniqueSignersCount: 1,
+        dataFeeds: ["ETH"],
+      },
+    );
+
+    // Checking expected amount
+    const expectedEthAmount = await wrappedMarketplaceContract.getPrice(orderId);
+
+    // Sending buy tx
+    const buyTx = await wrappedMarketplaceContract.wi(orderId, {
+      value: expectedEthAmount.mul(101).div(100), // a buffer for price movements
+    });
+    await buyTx.wait();
+
+    return buyTx;
+  } catch { 
+    const errText = "Error happened while buying the NFT";
+    alert(errText);
+  }
+}
+```
+
+### Test dApp locally
+
+#### 1. Check if tests pass
+```sh
+yarn test
+``` 
+
+#### 2. Compile contracts
+```sh
+yarn compile
```

Comment:
This command does not exist in `package.json`

Reply:
Fixed